File: testdata/point.js
/**
 * @constructor
 */
function Point(x, y) {
  this.x = x;
  this.y = y;
}

/** @type {number} */
Point.prototype.x;

/** @type {number} */
Point.prototype.y;

/**
 * @param {Point} other .
 */
Point.prototype.distanceTo = function(other) {
  var dx = this.x - other.x;
  var dy = this.y - other.y;
  return Math.sqrt(dx * dx + dy * dy);
};

File: testdata/point.dart
import 'dart:math';

class Point {
  final num x, y;

  Point(this.x, this.y);
  //Point.zero() : x = 0, y = 0;  // Named constructor are not supported

  num distanceTo(Point other) {
    var dx = x - other.x;
    var dy = y - other.y;
    return sqrt(dx * dx + dy * dy);
  }
}
